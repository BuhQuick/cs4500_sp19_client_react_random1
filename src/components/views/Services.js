import React from 'react'
const Services = ({services, onChangeItemsPerPage, nextPage, previousPage, createPageButtons}) =>
    {
        let div = 
            (
                <div>
                <h3>Services</h3>
                <table className="table">
                <tr>
                <th>Title</th>
                </tr>
                <tbody>
                {
                    services.map(service =>
                            <tr key={service.id}>
                                <td>{service.serviceName}</td>
                            </tr>)
                }
                </tbody>
                <tr>
                    <th><button id="previous" onClick={previousPage}>Previous</button></th>
                    <th><div id="pageButtons"> </div></th>
                    <th><button id="next" onClick={nextPage}>Next</button></th>
                </tr>
                <tr>
                    <th>
                      <div id = "dropdown" >
                          <select id="selector" onChange={onChangeItemsPerPage}>
                              <option value="1" selected="selected">10</option>
                              <option value="2">25</option>
                              <option value="3">50</option>
                              <option value="4">100</option>
                              <option value="5">All</option></select>
                      </div>
                    </th>
                </tr>
            </table>
                </div>
            )
            createPageButtons()
            return div;
    }

export default Services
