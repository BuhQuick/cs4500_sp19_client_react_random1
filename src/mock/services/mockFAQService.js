var faqData = require('../data/faq.mock.json')

global.fetch = jest.fn()
  .mockImplementation(url => {
    if(url.includes("/faqs/102")) {
      return new Promise((resolve,reject) => {
        resolve({json:function() {
          return {
            "id":102,
            "question": "What is my second question?",
            "title": "Q2"
          }
        }});
      });
    }
    else if(url.includes("/faqs") {
      return new Promise(resolve => resolve(faqsData))

    }
  });
