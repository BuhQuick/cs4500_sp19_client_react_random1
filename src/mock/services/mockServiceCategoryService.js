var srvCatData = require('../data/service_categories.mock.json')

global.fetch = jest.fn()
    .mockImplementation((url, config) => {
        if(!config) {
            if(url.indexOf('api/categories/0') != -1) { 
                return srvCatData[0]
            } else if(url.indexOf('api/categories/1') != -1) {
                return srvCatData[1]
            } else if(url.indexOf('api/categories') != -1) {
                return new Promise((resolve, reject) => {
                    resolve({ json: function() {
                        return srvCatData
                    }})})
            }
        }});
