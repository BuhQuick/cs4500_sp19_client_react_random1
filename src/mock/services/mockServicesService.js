var srvData = require('../data/service.mock.json')

let findAllServices = () => new Promise(resolve => resolve(srvData))

export default findAllServices
