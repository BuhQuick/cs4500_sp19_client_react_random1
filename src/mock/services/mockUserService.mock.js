import users from '../MockData/Users.mock'

global.fetch = jest.fn().mockImplementation((url, config) => {
    if (!config) {
        if (url.indexOf('api/users/1') != -1) {
            return users[0];
        } else if (url.indexOf('api/users/10') != -1) {
            return users[1];
        } else if (url.indexOf('api/users/11') != -1) {
            return users[2];
        } else if (url.indexOf('api/users/12') != -1) {
            return users[3];
        } else if (url.indexOf('api/users') != -1) {
            return new Promise((resolve, reject) => {
                resolve({
                    json: function () {
                        return users;
                    }
                })
            })
        }
    } else if (config.method === 'put') {
         if (url.indexOf('api/users/1') != -1) {
             users[0] = JSON.parse(config.body);
             return users[0];
         } else if (url.indexOf('api/users/10') != -1) {
             users[1] = JSON.parse(config.body);
             return users[1];
         } else if (url.indexOf('api/users/11') != -1) {
             users[2] = JSON.parse(config.body);
             return users[2];
         }else if (url.indexOf('api/users/12') != -1) {
             users[3] = JSON.parse(config.body);
             return users[3];
         }
    } else if (config.method === 'delete') {
          if (url.indexOf('api/users/1') != -1) {
              return users[0];
          } else if (url.indexOf('api/users/10') != -1) {
              return users[1];
          } else if (url.indexOf('api/users/11') != -1) {
              return users[2];
          } else if (url.indexOf('api/users/12') != -1) {
              return users[3];
          }
    } else if (config.method === 'post') {
        let user = JSON.parse(config.body);
        user.id = (new Date()).getTime();
        users.push(user);
        return new Promise((resolve, reject) => {
            resolve({
                json: function () {
                    return users;
                }
            })
        })
    }
});