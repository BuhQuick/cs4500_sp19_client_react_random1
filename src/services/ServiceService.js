import MIDDLE_TIER_URL from '../Env.js'
export default class ServiceService {
    static instance = null;
    static getInstance() {
        if(ServiceService.instance === null) {
            ServiceService.instance = new ServiceService()
        }
        return this.instance
    }
    findServiceById = serviceId =>
        fetch(MIDDLE_TIER_URL + "/api/services/" + serviceId)
            .then(response => response.json())
    findAllServices = () =>
        fetch(MIDDLE_TIER_URL + "/api/services")
            .then(response => response.json())
}
