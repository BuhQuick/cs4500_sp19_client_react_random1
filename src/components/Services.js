import React from 'react'
import ServiceService from '../services/ServiceService'
class Services extends React.Component {
    constructor(props) {
        super(props)
            this.serviceService = ServiceService.getInstance()
            this.state = {
                services: [],
                pageNumber: 1,
                itemsPerPage: 10
            }
    }
    componentDidMount() {
        this.serviceService
            .findAllServices()
            .then(results =>
                    this.setState({
                        services: results
                    })
                 )
    }
    onChangeItemsPerPage() {
        var selectorID = document.getElementById("selector");
        var pageText = selectorID.options[selectorID.selectedIndex].text;

        if (pageText === "10") {
            this.setState({itemsPerPage: 10, pageNumber: 1})
        }
        if (pageText === "25") {
            this.setState({itemsPerPage: 25, pageNumber: 1})
        }
        if (pageText === "50") {
            this.setState({itemsPerPage: 50, pageNumber: 1})
        }
        if (pageText === "All") {
            this.setState({itemsPerPage: -1, pageNumber: 1})
        }
    }
    previousPage() {
        if (this.state.pageNumber !== 1) {
            this.setState({pageNumber: (this.state.pageNumber - 1) })
        }
    }
    nextPage() {
        var totalPages = Math.ceil(this.state.services.length / this.state.itemsPerPage);
        if (this.state.pageNumber + 1 <= totalPages) {
            this.setState({pageNumber: (this.state.pageNumber + 1) })
        }
    }
    choosePage(newPage) {
        this.setState({pageNumber: newPage })

    }
    createPageButtons() {
        var totalPages = 1
        if (this.state.itemsPerPage !== -1) {
            totalPages = Math.ceil(this.state.services.length / this.state.itemsPerPage);
        }
        var i;
        for (i = 1; i < totalPages + 1; i++) {
            if ((document.getElementById("Button " + i)) === null)
            {
                console.log(i);
                var button = document.createElement("BUTTON");
                var buttonText = document.createTextNode("" + i);
                button.appendChild(buttonText);
                button.type = "button";
                button.id = "Button " + i;
                button.value = i;
                button.onclick = function(e) {
                    this.setState({pageNumber: e.target.value})
                }.bind(this);
                document.getElementById("pageButtons").appendChild(button);
            }
        }
    }
    render() {
        let div = 
            (
                <div>
                <h3>Services</h3>
                <table className="table">
                <tr>
                <th>Title</th>
                </tr>
                <tbody>
                {
                    this.state.services.slice(   ((this.state.pageNumber * this.state.itemsPerPage) - this.state.itemsPerPage), ((this.state.pageNumber * this.state.itemsPerPage) - 1))
                        .map(service =>
                            <tr key={service.id}>
                                <td>{service.serviceName}</td>
                                <td>{service.serviceCategories}</td>
                            </tr>
                        )
                }
                </tbody>
                <tr>
                    <th><button id="previous" onClick={() => this.previousPage()}>Previous</button></th>
                    <th><div id="pageButtons"> </div></th>
                    <th><button id="next" onClick={() => this.nextPage()}>Next</button></th>
                </tr>
                <tr>
                    <th>
                      <div id = "dropdown" >
                          <select id="selector" onChange={() => this.onChangeItemsPerPage()}>
                              <option value="1" selected="selected">10</option>
                              <option value="2">25</option>
                              <option value="3">50</option>
                              <option value="4">100</option>
                              <option value="5">All</option></select>
                      </div>
                    </th>
                </tr>
            </table>
                </div>
            )
            this.createPageButtons()
            return div;
    }
}

export default Services
