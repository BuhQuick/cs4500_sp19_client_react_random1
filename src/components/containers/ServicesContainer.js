import React from 'react'
import ServiceService from '../../services/ServiceService'
import Services from '../views/Services'
class ServicesContainer extends React.Component {
    constructor(props) {
        super(props)
        this.serviceService = this.props.service
        this.state = 
        { 
            services: [],
            pageNumber: 1,
            itemsPerPage: 10
        }
        this.onChangeItemsPerPage = this.onChangeItemsPerPage.bind(this)
        this.nextPage = this.nextPage.bind(this)
        this.previousPage = this.previousPage.bind(this)
        this.createPageButtons = this.createPageButtons.bind(this)
    }
    
    componentDidMount() {
        this.serviceService
            .findAllServices()
            .then(results =>
                    this.setState({
                        services: results
                    })
                 )
    }

    onChangeItemsPerPage = () =>
    {
        var selectorID = document.getElementById("selector");
        var pageText = selectorID.options[selectorID.selectedIndex].text;

        if (pageText === "10") {
            this.setState({itemsPerPage: 10, pageNumber: 1})
        }
        if (pageText === "25") {
            this.setState({itemsPerPage: 25, pageNumber: 1})
        }
        if (pageText === "50") {
            this.setState({itemsPerPage: 50, pageNumber: 1})
        }
        if (pageText === "All") {
            this.setState({itemsPerPage: -1, pageNumber: 1})
        }

    }

    previousPage = () =>
    {
        if (this.state.pageNumber !== 1) {
            this.setState({pageNumber: (this.state.pageNumber - 1) })
        }
    }

    nextPage = () =>
    {
        var totalPages = Math.ceil(this.state.services.length / this.state.itemsPerPage);
        if (this.state.pageNumber + 1 <= totalPages) {
            this.setState({pageNumber: (this.state.pageNumber + 1) })
        }
    }

    createPageButtons = () => 
    {
        var totalPages = 1
        if (this.state.itemsPerPage !== -1) {
            totalPages = Math.ceil(this.state.services.length / this.state.itemsPerPage);
        }
        var i;
        for (i = 1; i < totalPages + 1; i++) {
            if ((document.getElementById("Button " + i)) === null)
            {
                console.log(i);
                var button = document.createElement("BUTTON");
                var buttonText = document.createTextNode("" + i);
                button.appendChild(buttonText);
                button.type = "button";
                button.id = "Button " + i;
                button.value = i;
                button.onclick = function(e) {
                    this.setState({pageNumber: e.target.value})
                }.bind(this);
                document.getElementById("pageButtons").appendChild(button);
            }
        }
    }
    
    render() {
        return (
                <Services 
                services={this.state.services.slice(
                ((this.state.pageNumber * this.state.itemsPerPage) - this.state.itemsPerPage), 
                ((this.state.pageNumber * this.state.itemsPerPage) - 1)
                )}
                onChangeItemsPerPage={this.onChangeItemsPerPage}
                nextPage={this.nextPage}
                previousPage={this.previousPage}
                createPageButtons={this.createPageButtons}
                />)
    }
 
}
export default ServicesContainer;
