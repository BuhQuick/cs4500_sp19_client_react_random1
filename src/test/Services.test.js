import React from 'react';
import Services from '../components/views/Services';
import ServicesContainer from '../components/containers/ServicesContainer';
import renderer from 'react-test-renderer';
import ServiceService from '../services/ServiceService'
const serviceService = ServiceService.getInstance()
import '../mock/services/mockServicesService'

test('service list renders correctly', () => {
    const testRenderer = renderer.create(
            <ServicesContainer
                service={serviceService}/>)

        let tree = testRenderer.toJSON()
        expect(tree).toMatchSnapshot()
})

test('find all services', () => {
    return serviceService.findAllServices()
        .then(services => {
            expect(services).toBeDefined()
            expect(services).toHaveLength(4)
            expect(services[0].id).toBe(101)
        })
})
