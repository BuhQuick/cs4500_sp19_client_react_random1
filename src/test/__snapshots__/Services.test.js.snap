// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service list renders correctly 1`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="table"
  >
    <tr>
      <th>
        Title
      </th>
    </tr>
    <tbody />
    <tr>
      <th>
        <button
          id="previous"
          onClick={[Function]}
        >
          Previous
        </button>
      </th>
      <th>
        <div
          id="pageButtons"
        >
           
        </div>
      </th>
      <th>
        <button
          id="next"
          onClick={[Function]}
        >
          Next
        </button>
      </th>
    </tr>
    <tr>
      <th>
        <div
          id="dropdown"
        >
          <select
            id="selector"
            onChange={[Function]}
          >
            <option
              selected="selected"
              value="1"
            >
              10
            </option>
            <option
              value="2"
            >
              25
            </option>
            <option
              value="3"
            >
              50
            </option>
            <option
              value="4"
            >
              100
            </option>
            <option
              value="5"
            >
              All
            </option>
          </select>
        </div>
      </th>
    </tr>
  </table>
</div>
`;
